### DISCUSSION TITLE
FTDI select 3.3 volts with the switch  On the VCC output, 5 volts remain

### DISCUSSION THREAD

**Discussion by khaleids** [arendst/Tasmota#20457] [Q&A] [answered]
Hello everyone, I have a USB to Serial module
 The problem is that when you select 3.3 volts with the switch
 On the VCC output, 5 volts remain
 I bought a voltage regulator 8K1L 78c33 to reduce the voltage, but I do not know how to connect it and there is no data sheet for it.

 The question is how to know the characteristic of every pin?
 How to connect it to the circuit to output 3.3 volts instead of 5 volts?
I want to use it with Sonoff Basic R2 Esp8285

https://ae01.alicdn.com/kf/Sebd22c66733d4af6a15ac699b0f4dcefw.jpg_640x640Q90.jpg_.webp

![FTDI](https://github.com/arendst/Tasmota/assets/23697904/658343ce-9b7d-421e-b1d9-2416c456d30a)
![VR](https://github.com/arendst/Tasmota/assets/23697904/1bdeeb75-dc72-409e-9a7e-125a7943a1e6)

**Comment by Jason2866** [comment, marked as answer]
Short answer. Do not use this typ of serial usb converter for flashing esp devices.
Use the recommended ones. https://tasmota.github.io/docs/Getting-Started/#serial-programmer

**Reply by sfromis** [reply]
What might work with that dubious thing would be to not at all use the voltage pins from it, but instead supply voltage to the device to be flashed using some good external 3.3V power source, with common ground for all, and still connect tx/rx. And of course with the switch set to 3.3V still.

**Comment by lalo-uy** [comment]
This is the regulator datasheet.
https://www.s-manuals.com/pdf/datasheet/n/j/njm78l02_jrc.pdf

But wil not be any good for flashing, since is only 100mA.

**Reply by sfromis** [reply]
While 100 mA is indeed not "good" for flashing, it *may* still work for flashing ESP8266. At least 150 mA recommended.
